driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -366,10 +366,9 @@ static boolean canRetryRead(final ServerDescription serverDescription, final Con
 
     private static <T> QueryResult<T> cursorDocumentToQueryResult(final BsonDocument cursorDocument, final ServerAddress serverAddress,
                                                                   final String fieldNameContainingBatch) {
-        long cursorId = ((BsonInt64) cursorDocument.get("id")).getValue();
```

Comment:
This file can be reverted back to as it is on master.

Reply:
Done.